src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
+
+  return sum / women.length;
```

Comment:
Read previous comment
```suggestion
  const women = withChildren
    ? people.filter(({ sex, name }) => 
         sex === 'f' && 
         people.find(({ mother: motherName }) => mother === name)
      )
    : people.filter(({ sex }) => sex === 'f');

  const sum = men.reduce((acc, { died, born }) => {
     acc += died - born;
     
     return acc;
  }, 0);

  return sum / women.length;
```